exotica/src/Scene.cpp

```diff
@@ -689,6 +689,15 @@ void Scene::addObject(const std::string& name, const KDL::Frame& transform, cons
     if (updateCollisionScene) updateCollisionObjects();
 }
 
+void Scene::addObjectToEnvironment(const std::string& name, const KDL::Frame& transform, shapes::ShapeConstPtr shape, const bool updateCollisionScene)
+{
+    Eigen::Isometry3d pose;
+    tf::transformKDLToEigen(transform, pose);
+    ps_->getWorldNonConst()->addToObject(name, shape, pose);
```

Comment:
We shouldn't resort to the segway of the MoveIt planning scene here - you can directly call the KinematicTree method to add it (``KinematicTree::AddEnvironmentElement``)

Reply:
This is actually the same procedure that is used when loading the scene via one of the scene files:
[`world_->addToObject(ns, shapes::ShapePtr(s), pose);`](https://github.com/ros-planning/moveit/blob/1812bdb050e417e2c8591a82eaf72f993b0ba886/moveit_core/planning_scene/src/planning_scene.cpp#L1101)

Using moveit's `addToObject` is more in line with `loadGeometryFromStream`. What would be the advantage of using `KinematicTree::AddEnvironmentElement`?